cmd/notation/plugin/install.go

```diff
@@ -0,0 +1,329 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/notaryproject/notation-go/dir"
+	"github.com/notaryproject/notation-go/log"
+	notationplugin "github.com/notaryproject/notation/cmd/notation/internal/plugin"
+	"github.com/notaryproject/notation/internal/cmd"
+	"github.com/notaryproject/notation/internal/osutil"
+	"github.com/spf13/cobra"
+)
+
+const (
+	notationPluginTmpDir = "notationPluginTmpDir"
+)
+
+type pluginInstallOpts struct {
+	cmd.LoggingFlagOpts
+	pluginSourceType notationplugin.PluginSourceType
+	pluginSource     string
+	inputCheckSum    string
```

Comment:
Replace all `checkSum` to `checksum` with lower-case `s`. See https://en.wikipedia.org/wiki/Checksum